shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -250,19 +250,38 @@ Engine::Engine(Delegate& delegate,
   // Setup the callback that will instantiate the rasterizer.
   flutter::Shell::CreateCallback<flutter::Rasterizer> on_create_rasterizer;
 #if defined(LEGACY_FUCHSIA_EMBEDDER)
-  on_create_rasterizer = [this](flutter::Shell& shell) {
+  on_create_rasterizer = [this, &product_config](flutter::Shell& shell) {
     if (use_legacy_renderer_) {
       FML_DCHECK(session_connection_);
       FML_DCHECK(surface_producer_);
       FML_DCHECK(legacy_external_view_embedder_);
 
+      if (product_config.enable_shader_warmup()) {
+        FML_DCHECK(surface_producer_);
+        WarmupSkps(shell.GetDartVM()
+                       ->GetConcurrentMessageLoop()
+                       ->GetTaskRunner()
+                       .get(),
+                   shell.GetTaskRunners().GetRasterTaskRunner().get(),
+                   surface_producer_.value());
+      }
+
       auto compositor_context =
           std::make_unique<flutter_runner::CompositorContext>(
               session_connection_.value(), surface_producer_.value(),
               legacy_external_view_embedder_);
       return std::make_unique<flutter::Rasterizer>(
           shell, std::move(compositor_context));
     } else {
+      if (product_config.enable_shader_warmup()) {
```

Comment:
This should be below, after the #else

If you follow both of these suggestions the warmup will work in all cases and you won't need 3 if blocks

Reply:
I have one already on the other side of the #else and i cant put it outside of the #if/endif block because that encapsulates the entire on_create_rasterizer lambda, and this needs to be inside that lambda in order to have access to the fully constructed Shell (unless theres somewhere better to put this that has access to a fully constructed Shell?)

Reply:
Oh, duh...I totally missed that while reviewing.  Ignore this comment